### PR TITLE
Add API and related terms

### PR DESCRIPTION
**Pull request by franknoirot**
Reviewers are optional if you're interested in looking these over, no rush!

Adds entries for:
- API
- host
- client
- endpoint

and adds more synonyms for existing posts. I think we should make a rule to include **plurals** of any terms in the synonyms, because I want us to be able to wrap the entire term when we add the website term scanning functionality, even with compound plurals like "application programming interfaces".

### CODE REVIEW

File: content/glossary/endpoint.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    - methods
+    - service
+    - services
+excerpt: A single callable service within an API.
+domains:
+    - software-engineering
+---
+When discussing [APIs](/glossary/application-programming-interface-api), an "endpoint" is a single service that is offered by a [host](/glossary/host) software that can be requested by a [client](/glossary/client). An endpoint on an API hosted on the internet is usually represented by a URL, such as `https://api.kittycad.io/user`.
+
+Some endpoints may ask the host fetch and return some data, like [getting your user](https://kittycad.io/docs/api/get-your-user), or ask them to perform some work for the client and return the result, like [converting a CAD file](https://kittycad.io/docs/api/convert-cad-file).
```
Comment (Irev-Dev) [inline content/glossary/endpoint.mdx:15]: `Some endpoints may ask the host fetch and return some data` I feel like something's going awry here, is it meant to be "ask the host **to** fetch and"

File: content/glossary/host.mdx
Comment (Irev-Dev) [inline content/glossary/host.mdx:1]: Should this have a synonym of server?

Comment (franknoirot) [reply content/glossary/host.mdx:1]: Yeah sure, I wasn't sure if that should be its own entry but that makes sense.

**Comment by franknoirot** [comment]
Got @Irev-Dev's feedback addressed.

**Review by Irev-Dev — APPROVED** [review]

**Pull request merged**